src/utility/index.jsx

```diff
@@ -592,3 +592,47 @@ export const wilsonScore = (up, down) => {
     1 + (z * z / n)
   );
 };
+
+// example: Radiant Top Tower Tier 1
+export const translateBuildings = {
+  radiant: {
+    npc_dota_goodguys_fort: `${strings.general_radiant} ${strings.building_ancient}`,
+    npc_dota_goodguys_healers: `${strings.general_radiant} ${strings.heading_shrine}`,
+    npc_dota_goodguys_tower1_top: `${strings.general_radiant} ${strings.top_tower} ${strings.tier1}`,
+    npc_dota_goodguys_tower2_top: `${strings.general_radiant} ${strings.top_tower} ${strings.tier2}`,
+    npc_dota_goodguys_tower3_top: `${strings.general_radiant} ${strings.top_tower} ${strings.tier3}`,
+    npc_dota_goodguys_tower1_mid: `${strings.general_radiant} ${strings.mid_tower} ${strings.tier1}`,
+    npc_dota_goodguys_tower2_mid: `${strings.general_radiant} ${strings.mid_tower} ${strings.tier2}`,
+    npc_dota_goodguys_tower3_mid: `${strings.general_radiant} ${strings.mid_tower} ${strings.tier3}`,
+    npc_dota_goodguys_tower1_bot: `${strings.general_radiant} ${strings.bot_tower} ${strings.tier1}`,
+    npc_dota_goodguys_tower2_bot: `${strings.general_radiant} ${strings.bot_tower} ${strings.tier2}`,
+    npc_dota_goodguys_tower3_bot: `${strings.general_radiant} ${strings.bot_tower} ${strings.tier3}`,
+    npc_dota_goodguys_tower4: `${strings.general_radiant} ${strings.heading_tower} ${strings.tier4}`,
+    npc_dota_goodguys_melee_rax_top: `${strings.barracks_value_1024} ${strings.heading_barracks}`,
+    npc_dota_goodguys_melee_rax_mid: `${strings.barracks_value_256} ${strings.heading_barracks}`,
+    npc_dota_goodguys_melee_rax_bot: `${strings.barracks_value_64} ${strings.heading_barracks}`,
+    npc_dota_goodguys_range_rax_top: `${strings.barracks_value_2048} ${strings.heading_barracks}`,
+    npc_dota_goodguys_range_rax_mid: `${strings.barracks_value_512} ${strings.heading_barracks}`,
+    npc_dota_goodguys_range_rax_bot: `${strings.barracks_value_128} ${strings.heading_barracks}`,
+  },
+  dire: {
```

Comment:
since the only thing different about the dire strings is that the first part is different, you can probably cut the code in half by making this a function that takes a param for radiant/dire

Reply:
There is no radiant/dire value I could pass to a function. I use the translateBuildings object itself to figure out if it's radiant or dire.

I thought about it for 20 minutes but I can't figure out how to shorten the code :(

Reply:
Don't you have row.isRadiant on the row?